src/lib.rs

```diff
@@ -235,25 +232,46 @@ where
     ///
     /// Then copy the `CAN_BUS_TIME` register value from the table and pass it as the `btr`
     /// parameter to this method.
-    pub fn set_bit_timing(&mut self, btr: u32) {
+    pub fn set_bit_timing(&mut self, btr: u32) -> &mut Self {
         let can = self.registers();
         can.btr.modify(|r, w| unsafe {
             let mode_bits = r.bits() & 0xC000_0000;
             w.bits(mode_bits | btr)
         });
+        self
     }
 
     /// Enables or disables loopback mode: Internally connects the TX and RX
     /// signals together.
-    pub fn set_loopback(&mut self, enabled: bool) {
+    pub fn set_loopback(&mut self, enabled: bool) -> &mut Self {
         let can = self.registers();
         can.btr.modify(|_, w| w.lbkm().bit(enabled));
+        self
     }
 
     /// Enables or disables silent mode: Disconnects the TX signal from the pin.
-    pub fn set_silent(&mut self, enabled: bool) {
+    pub fn set_silent(&mut self, enabled: bool) -> &mut Self {
         let can = self.registers();
         can.btr.modify(|_, w| w.silm().bit(enabled));
+        self
+    }
+}
+
+impl<I: Instance> Drop for CanConfig<'_, I> {
+    #[inline]
+    fn drop(&mut self) {
+        /*
+            sleep implementation
+        */
```

Comment:
This seems incorrect

```suggestion
        // Leave initialization mode
```